app_grim_reefer/src/main.c

```diff
@@ -27,54 +47,168 @@ const struct gpio_dt_spec led2 = GPIO_DT_SPEC_GET(LED2_NODE, gpios);
+const struct gpio_dt_spec buzzer = GPIO_DT_SPEC_GET(BUZZER_NODE, gpios);
+
+#define DBG_SERIAL_NODE DT_ALIAS(debug_serial)
+const struct device *const debug_serial_dev = DEVICE_DT_GET(DBG_SERIAL_NODE);
+
+#define BME280_NODE DT_NODELABEL(bme280)
+const struct device *bme280_dev = DEVICE_DT_GET(BME280_NODE);
+
+#define LSM6DSL_NODE DT_NODELABEL(lsm6dsl)
+const struct device *lsm6dsl_dev = DEVICE_DT_GET(LSM6DSL_NODE);
+
+#define FLASH_NODE DT_NODELABEL(w25q512)
+const struct device *flash_dev = DEVICE_DT_GET(FLASH_NODE);
+
+// Inas
+#define INA_BAT_NODE DT_NODELABEL(ina260_battery)
+const struct device *ina_bat_dev = DEVICE_DT_GET(INA_BAT_NODE);
+
+#define INA_LDO_NODE DT_NODELABEL(ina260_ldo)
+const struct device *ina_ldo_dev = DEVICE_DT_GET(INA_LDO_NODE);
+
+#define INA_GRIM_NODE DT_NODELABEL(ina260_3v3)
+const struct device *ina_grim_dev = DEVICE_DT_GET(INA_GRIM_NODE);
+
+static const struct adc_dt_spec adc_chan0 = ADC_DT_SPEC_GET_BY_IDX(DT_PATH(zephyr_user), 0);
+
+#define INIT_GPIO_FAIL      -1
+#define INIT_NOFLASH        -1
+#define INIT_MISSING_SENSOR -2
+#define INIT_OK             0
+
+static int gpio_init(void) {
     // Init LEDS
     if (!gpio_is_ready_dt(&led1)) {
         LOG_ERR("LED 1 is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&led1, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure LED 1 output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
+
     if (!gpio_is_ready_dt(&led2)) {
         LOG_ERR("LED 2 is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&led2, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure LED 2 output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     // Init Enable pins
     if (!gpio_is_ready_dt(&ldo_enable)) {
         LOG_ERR("ldo enable pin is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&ldo_enable, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure ldo enable output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
+
     if (!gpio_is_ready_dt(&cam_enable)) {
         LOG_ERR("camera enable pin is not ready\n");
-        return -1;
+        return INIT_GPIO_FAIL;
     }
     if (gpio_pin_configure_dt(&cam_enable, GPIO_OUTPUT_ACTIVE) < 0) {
         LOG_ERR("Unable to configure camera enable output pin\n");
-        return -1;
+        return INIT_GPIO_FAIL;
+    }
+
+    if (!gpio_is_ready_dt(&buzzer)) {
+        LOG_ERR("buzzer pin is not ready\n");
+        return INIT_GPIO_FAIL;
+    }
+    if (gpio_pin_configure_dt(&buzzer, GPIO_OUTPUT_ACTIVE) < 0) {
+        LOG_ERR("Unable to configure buzzer output pin\n");
+        return INIT_GPIO_FAIL;
+    }
+
+    if (!device_is_ready(debug_serial_dev)) {
+        LOG_ERR("Debug serial not ready\n");
+        return INIT_GPIO_FAIL;
     }
```

Comment:
Think you can make this a helper function giving it a gpio_dt_spec and a string of the device name. Pretty sure these devices also have names too in their struct, so you may not even need to pass a string arg.

Reply:
Good idea :+1: